x/execution/internal/keeper/keeper.go

```diff
@@ -470,6 +407,62 @@ func (k *Keeper) fetchEmitters(ctx sdk.Context, proc *process.Process, nodeKey s
 	return matchedRuns, nil
 }
 
+func (k *Keeper) calculatePrice(ctx sdk.Context, exec *executionpb.Execution) (sdk.Int, error) {
+	inst, err := k.instanceKeeper.Get(ctx, exec.InstanceHash)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	srv, err := k.serviceKeeper.Get(ctx, inst.ServiceHash)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	task, err := srv.GetTask(exec.TaskKey)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	perCall := sdk.NewInt(200)
+	if task.Price != nil && task.Price.PerCall != "" {
+		price, ok := sdk.NewIntFromString(task.Price.PerCall)
+		if !ok {
+			return sdk.Int{}, fmt.Errorf("invalid price per call %q", task.Price.PerCall)
+		}
+		if price.GT(perCall) {
+			perCall = price
+		}
+	}
+	perSec := sdk.NewInt(20000)
+	if task.Price != nil && task.Price.PerSec != "" {
+		price, ok := sdk.NewIntFromString(task.Price.PerSec)
+		if !ok {
+			return sdk.Int{}, fmt.Errorf("invalid price per sec %q", task.Price.PerSec)
+		}
+		if price.GT(perSec) {
+			perSec = price
+		}
+	}
+	perKB := sdk.NewInt(100)
+	if task.Price != nil && task.Price.PerKB != "" {
+		price, ok := sdk.NewIntFromString(task.Price.PerKB)
+		if !ok {
+			return sdk.Int{}, fmt.Errorf("invalid price per kb %q", task.Price.PerKB)
+		}
+		if price.GT(perKB) {
+			perKB = price
+		}
```

Comment:
didn't we say we could put whatever value we want? like 0 credit per KB but a lot per call?

Reply:
If we don't check for minimum value then this is just a default and any user can go bellow this. This is a minimal value and not a default value but nothing stop us to put a minimal value equals to 0 and then have the behavior that you describe.

Another way is to set this `price` field require and have the check on the `service.create` only but this is a massive breaking change as all services need to be redeployed.

Reply:
After discussion, we will not put any limitation on the price, no default, and no minimum limit. Price is now required for all tasks so we can simplify this code.